_posts/2020-06-08-hhvm-4.61.markdown

```diff
@@ -0,0 +1,25 @@
+---
+title: "HHVM 4.61"
+layout: post
+author: jjergus
+category: blog
+---
+
+HHVM 4.61 is released! This release marks the end of support for 4.55;
+HHVM 4.56&ndash;4.60 remain supported, as does the 4.32 LTS release.
+
+# Highlights
+
+- The typechecker no longer reports multiple redundant errors when a class
+  (interface) is extended (implemented) with incompatible generic type
+  parameters
+  ([example](https://github.com/facebook/hhvm/blob/ba6a6cb57eabd7135e5aeb169c306d53b454dc44/hphp/hack/test/typecheck/constraints/extend_generic2.php)).
+- Fixed a bug where using `array_unshift` on a `keyset` could cause the `keyset`
+  to contain multiple identical values.
+
+# Future Changes
+
+- Support for `HH_IGNORE_ERROR` may be removed in the future, making `HH_FIXME`
+  the only way to silence typechecker errors.
```

Comment:
Perhaps something like:

If you are using these with different intent, consider replacing `/* HH_IGNORE_ERROR[1234] message */` with `/* HH_FIXME[1234] IGNORE_ERROR message */` to make searching simpler.

Reply:
Hmm that seems like a needlessly specific recommendation, it seems better to me to just let people figure out what works for them...

Also from what I remember, HH_IGNORE_ERROR was never really meant to silence typechecker errors, so hopefully not many people are using it for that. It was added when we migrated some linters to hack-lint, since it felt weird to use FIXME for intentional lint errors.

Reply:
We explicitly discuss the distinction here: https://docs.hhvm.com/hack/silencing-errors/introduction#hh_fixme-versus-hh_ignore_error